src/Symfony/Component/Translation/Translator.php

```diff
@@ -307,9 +351,13 @@ public function getMessages($locale = null)
 
     /**
      * @param string $locale
+     *
+     * @deprecated since version 2.8, to be removed in 3.0. Rely on CatalogueBagInterface::getCatalogue instead.
      */
     protected function loadCatalogue($locale)
     {
+        @trigger_error('The '.__METHOD__.' method is deprecated since version 2.8 and will be removed in 3.0. Rely on CatalogueBagInterface::getCatalogue() instead.', E_USER_DEPRECATED);
```

Comment:
fixed and here is how it was solved https://github.com/aitboudad/symfony/blob/translation_loader_resolver/src/Symfony/Component/Translation/Translator.php#L303-L324